internal/ir/xds.go

```diff
@@ -384,3 +395,41 @@ func (s StringMatch) Validate() error {
 
 	return errs
 }
+
+// TCPListener holds the listener configuration.
+// +k8s:deepcopy-gen=true
+type TCPListener struct {
+	// Name of the TCPListener
+	Name string
+	// Address that the listener should listen on.
+	Address string
+	// Port on which the service can be expected to be accessed by clients.
+	Port uint32
+	// Server names that are compared against the server names of a new connection.
+	// Wildcard hosts are supported in the prefix form. Partial wildcards are not
+	// supported, and values like *w.example.com are invalid.
+	// SNIs are used only in case of TLS Passthrough.
+	SNIs []string
```

Comment:
if we added SNIs under a TLS struct, we could validate it to ensure that SNIs is not empty